include/deal.II/lac/scalapack.h

```diff
@@ -389,6 +546,24 @@ class ScaLAPACKMatrix : protected TransposeTable<NumberType>
    */
   NumberType &local_el(const unsigned int loc_row, const unsigned int loc_column);
 
+  /**
+   * Scale the columns of the distributed matrix by scalars in array @p factors.
+   *
+   * The array @p factors must have as many entries as the matrix columns.
+   *
+   * Copies of @p factors have to available on all processes of the underlying MPI communicator
+   */
+  void scale_columns(std::vector<NumberType> &factors);
+
+  /**
+   * Scale the rows of the distributed matrix by scalars in array @p factors.
+   *
+   * The array @p factors must have as many entries as the matrix rows.
+   *
+   * Copies of @p factors have to available on all processes of the underlying MPI communicator
+   */
+  void scale_rows(std::vector<NumberType> &factors);
```

Comment:
same here

Reply:
btw, we should probably use `Vector<NumberType>` instead here. It won't make much of a difference at the end of the day, but is probably more consistent representation of a diagonal matrix used here as pre/post multiplication.

Reply:
we may seek just a few eigenvalues or all eigenvalues of a matrix. In other places (like Arpack for sparse large matrices) we also use `std::vector<>` which is problably ok.

Here, however, this vector has to have as many elements as matrix rows/columns. So in case of a square matrix, this is simply a diagonal matrix represented by a vector. With other matrix classes AFAIK we never use `std::vector<>` to represent a diagonal, i.e. https://www.dealii.org/developer/doxygen/deal.II/classSparseMatrix.html#a07ba942c5b7466b3895fed3104fbcc26